### PR TITLE
extend collector tests

### PR DESCRIPTION
**Pull request by lwsanty**
add cases with handling timeouts, immediate stops and send metrics errors

Signed-off-by: lwsanty <lwsanty@gmail.com>

### CODE REVIEW

File: metrics/metrics_test.go [hunk 1]
```diff
@@ -161,3 +163,97 @@ func TestMetricsCollectorByOrg(t *testing.T) {
...
+}
+
+func TestClosesWithDelay(t *testing.T) {
+	for _, c := range []closeDelayCase{
+		{"ImmediateWithoutDelay", true, time.Second, 0, 0},
+		{"ImmediateWithDelay", true, 500 * time.Millisecond, time.Second, 1},
+		{"NonImmediateWithoutDelay", false, time.Second, 0, 1},
+		{"NonImmediateWithDelay", false, 500 * time.Millisecond, time.Second, 2},
+	} {
+		c := c
```
Comment (jfontan) [inline metrics/metrics_test.go]: Why this variable copy/shadowing?

Comment (lwsanty) [reply metrics/metrics_test.go]: probably it's a habit left after [scopelint](https://github.com/kyoh86/scopelint) :)

Comment (jfontan) [reply metrics/metrics_test.go]: I didn't know about that linter :+1:

Comment (lwsanty) [reply metrics/metrics_test.go]: this one is a part of [golangci-lint](https://github.com/golangci/golangci-lint)
we could probably use it if we have a need in linters, however we should consider that they consume too much RAM

**Pull request merged**